src/components/FootprintDialog.tsx

```diff
@@ -0,0 +1,361 @@
+                  ) : (
+                    <Copy className="h-4 w-4" />
+                  )}
+                </Button>
+              </div>
+            </div>
+            <div>
+              <label className="text-sm font-medium">Number of Pins</label>
+              <Input
+                type="number"
+                min="1"
+                max="10000"
+                onChange={(e) => {
+                  let value = Number(e.target.value)
+                  if (value < 1) value = 0
+                  if (value > 4000) value = 4000
+                  if (footprinterName.match(/\d+$/)) return
+                  const newName = `${footprinterName}${value ? value : ""}`
+                  try {
+                    const newParams = fp.string(newName).json()
+                    updateFootprintString(newName, newParams)
+                  } catch (error) {
+                    console.error("Error updating footprint string:", error)
+                    setFootprinterString(newName)
+                    handleFootprinterPreview(newName)
+                  }
+                }}
+                placeholder="Enter number of pins..."
+                className="mt-1"
+              />
+            </div>
+            {params && (
+              <div className="space-y-2">
+                <label className="text-sm font-medium">Parameters</label>
+                {Object.entries(params)
+                  .filter(([key]) => key !== "fn")
+                  .map(([key]) => (
+                    <div key={key} className="flex gap-2 items-center">
+                      <label className="text-sm">{key}:</label>
+                      {typeof params[key] === "boolean" ? (
+                        <input
+                          type="checkbox"
+                          checked={params[key]}
+                          onChange={(e) => updateParam(key, e.target.checked)}
+                          className="h-4 w-4"
+                        />
+                      ) : (
+                        <Input
+                          type={
+                            typeof params[key] === "number" ? "number" : "text"
+                          }
+                          value={params[key]}
+                          onChange={(e) => updateParam(key, e.target.value)}
+                          className="flex-1"
+                        />
+                      )}
+                    </div>
+                  ))}
+              </div>
+            )}
+            <Button
+              onClick={() => {
+                handleInsertFootprint()
+                onOpenChange(false)
+              }}
+              disabled={!footprinterString || !chipName}
+              className="w-full"
+            >
+              Insert Footprint
+            </Button>
+          </div>
+          <div className="rounded-xl overflow-hidden w-[800px] h-[600px]">
+            {previewSvg && <div>{parse(previewSvg)}</div>}
```

Comment:
You can just use `_dangerouslySetInnterHtml`- this is probably more efficient than adding an expensive dependency